src/plugins/vis_type_vega/public/vega_view/vega_map_view/view.ts

```diff
@@ -0,0 +1,181 @@
+
+import './vega_map_view.scss';
+
+export class VegaMapView extends VegaBaseView {
+  private mapServiceSettings: MapServiceSettings = getMapServiceSettings();
+  private mapStyle = this.getMapStyle();
+
+  private getMapStyle() {
+    const { mapStyle } = this._parser.mapConfig;
+
+    return mapStyle === 'default' ? this.mapServiceSettings.defaultTmsLayer() : mapStyle;
+  }
+
+  private get shouldShowZoomControl() {
+    return Boolean(this._parser.mapConfig.zoomControl);
+  }
+
+  private getMapParams(defaults: { maxZoom: number; minZoom: number }): Partial<MapboxOptions> {
+    const { longitude, latitude, scrollWheelZoom } = this._parser.mapConfig;
+    const zoomSettings = validateZoomSettings(this._parser.mapConfig, defaults, this.onWarn);
+
+    return {
+      ...zoomSettings,
+      center: [longitude, latitude],
+      scrollZoom: scrollWheelZoom,
+    };
+  }
+
+  private async initMapContainer(vegaView: vega.View) {
+    let style: Style = defaultMabBoxStyle;
+    let customAttribution: MapboxOptions['customAttribution'] = [];
+    const zoomSettings = {
+      minZoom: defaultMapConfig.minZoom,
+      maxZoom: defaultMapConfig.maxZoom,
+    };
+
+    if (this.mapStyle && this.mapStyle !== userConfiguredLayerId) {
+      const tmsService = await this.mapServiceSettings.getTmsService(this.mapStyle);
+
+      if (!tmsService) {
+        this.onWarn(
+          i18n.translate('visTypeVega.mapView.mapStyleNotFoundWarningMessage', {
+            defaultMessage: '{mapStyleParam} was not found',
+            values: { mapStyleParam: `"mapStyle":${this.mapStyle}` },
+          })
+        );
+        return;
+      }
+      zoomSettings.maxZoom = (await tmsService.getMaxZoom()) ?? defaultMapConfig.maxZoom;
+      zoomSettings.minZoom = (await tmsService.getMinZoom()) ?? defaultMapConfig.minZoom;
+      customAttribution = getAttributionsForTmsService(tmsService);
+      style = (await tmsService.getVectorStyleSheet()) as Style;
+    } else {
+      customAttribution = this.mapServiceSettings.config.tilemap.options.attribution;
+    }
+
+    // In some cases, Vega may be initialized twice, e.g. after awaiting...
+    if (!this._$container) return;
+
+    const mapBoxInstance = new Map({
+      style,
+      customAttribution,
+      container: this._$container.get(0),
+      ...this.getMapParams({ ...zoomSettings }),
+    });
+
+    const initMapComponents = () => {
+      this.initControls(mapBoxInstance);
+      this.initLayers(mapBoxInstance, vegaView);
+
+      this._addDestroyHandler(() => {
+        if (mapBoxInstance.getLayer(vegaLayerId)) {
+          mapBoxInstance.removeLayer(vegaLayerId);
+        }
+        if (mapBoxInstance.getLayer(userConfiguredLayerId)) {
+          mapBoxInstance.removeLayer(userConfiguredLayerId);
+        }
+        mapBoxInstance.remove();
+      });
+    };
+
+    mapBoxInstance.once('load', initMapComponents);
+  }
+
+  private async updateVegaView(mapBoxInstance: Map, vegaView: vega.View) {
```

Comment:
this doesn't reference this, so need for this to be a method.

Reply:
done